src/server/session.ts

```diff
@@ -120,6 +120,117 @@ namespace ts.server {
         export const Unknown = "unknown";
     }
 
+    namespace Metrics {
+        let eventCounts: Map<number> = {};
+        let properties: Map<string> = {};
+        let dtsCount = 0;
+
+        const sendInterval = 1000 * 60 * 5; // 5 minutes
+        let nextSendTimeMs = Date.now() + sendInterval;
+
+        const settingNames = [

```

Comment:
do not redefine these. use `ts.optionDeclarations` instead, this is the main place for all accepted commandline options.